BaseClasses.py

```diff
@@ -88,7 +87,7 @@ def __init__(self, players: int):
         self.shuffle_ganon = True
         self.spoiler = Spoiler(self)
         self.fix_trock_doors = self.AttributeProxy(
-            lambda player: self.shuffle[player] != 'vanilla' or self.mode[player] == 'inverted')
+            lambda player: self.shuffle[player] != 'vanilla' or self.world_state[player] == 2)
```

Comment:
I don't like using ints here, same for item_functionality. Shouldn't that be a regular Options.Choice?
I'm guessing the problem is that this is still in core?